secboot/secboot_tpm.go

```diff
@@ -21,19 +21,50 @@
 package secboot
 
 import (
+	"errors"
 	"fmt"
 
 	sb "github.com/snapcore/secboot"
 
+	"github.com/snapcore/snapd/bootloader/efi"
 	"github.com/snapcore/snapd/logger"
 )
 
+var sbConnectToDefaultTPM = sb.ConnectToDefaultTPM
+
 func CheckKeySealingSupported() error {
-	logger.Noticef("checking TPM device availability...")
-	tconn, err := sb.ConnectToDefaultTPM()
+	logger.Noticef("checking if secure boot is enabled...")
+	if err := checkSecureBootEnabled(); err != nil {
+		logger.Noticef("secure boot not enabled: %v", err)
+		return err
+	}
+	logger.Noticef("secure boot is enabled")
+
+	logger.Noticef("checking if TPM device is available...")
+	tconn, err := sbConnectToDefaultTPM()
 	if err != nil {
-		return fmt.Errorf("cannot connect to TPM device: %v", err)
+		err = fmt.Errorf("cannot connect to TPM device: %v", err)
+		logger.Noticef("%v", err)
+		return err
 	}
 	logger.Noticef("TPM device detected")
 	return tconn.Close()
 }
+
+func checkSecureBootEnabled() error {
+	b, _, err := efi.ReadVarBytes("SecureBoot-8be4df61-93ca-11d2-aa0d-00e098032b8c")
```

Comment:
A comment where this comes from might be nice

Reply:
also it would be nice to have a comment on who the vendor ID here is, but I couldn't find who it was from in my cursory googling, so it's not a blocker to know the vendor ID exactly here

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/linux/efi.h?h=v5.4.32 has it as

EFI_GLOBAL_VARIABLE_GUID

Reply:
so I double checked, it's actually specified by UEFI itself

Reply:
thanks

Reply:
Added a comment about the EFI Global Variable vendor GUID